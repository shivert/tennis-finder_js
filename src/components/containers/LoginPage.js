import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { Row, Col, Layout, Card } from "antd";
import * as actions from "../../actions/loginActions";
import LoginForm from "../Login/LoginForm";

const { Header, Content } = Layout;
class LoginPage extends React.Component {
  render() {
    return (
      <div>
        <Layout style={{ minHeight: "100vh" }}>
          <Header style={{ background: "rgba(0,0,0,0)" }} />
          <Content style={{ margin: "0 16px" }}>
            <Row type="flex" justify="center">
              <Col span={6}>
                <h1>Tennis Finder</h1>
              </Col>
            </Row>
            <Row type="flex" justify="center">
              <Col span={6}>
                <Card title="Login">
                  <LoginForm />
                </Card>
              </Col>
            </Row>
          </Content>
        </Layout>
      </div>
    );
  }
}

LoginPage.propTypes = {
  actions: PropTypes.object.isRequired
};

function mapStateToProps(state) {
  return {
    fuelSavings: state.fuelSavings
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(actions, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);
