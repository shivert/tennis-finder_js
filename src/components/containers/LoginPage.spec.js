import React from "react";
import { mount, shallow } from "enzyme";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import { create, renderer } from "react-test-renderer";
import initialState from "../../reducers/initialState";
import LoginPage from "./LoginPage";
import { Button } from "antd";

describe("<LoginPage />", () => {
  it("renders correctly", () => {
    const tree = create(
      <Button type="primary" htmlType="submit" className="login-form-button">
        Login
      </Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
