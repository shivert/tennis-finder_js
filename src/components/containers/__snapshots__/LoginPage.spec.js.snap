// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> renders correctly 1`] = `
<button
  className="ant-btn login-form-button ant-btn-primary"
  onClick={[Function]}
  type="submit"
>
  <span>
    Login
  </span>
</button>
`;
